MapboxVision/Services/VideoPlayer.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  VideoPlayer.swift
+//  MapboxVision
+//
+//  Created by Alexander Pristavko on 5/17/19.
+//  Copyright © 2019 Mapbox. All rights reserved.
+//
+
+import Foundation
+import CoreMedia
+import AVFoundation
+
+protocol VideoPlayable: VideoSource {
+    var delegate: VideoPlayerDelegate? { get set }
+
+    func start()
+    func stop()
+}
+
+protocol VideoPlayerDelegate: AnyObject {
+    func playbackDidFinish()
+}
+
+private enum PlaybackConstants {
+    static let frameDuration: TimeInterval = 0.03
+}
+
+final class VideoPlayer: NSObject, VideoPlayable {
+    weak var delegate: VideoPlayerDelegate?
+
+    private var isPlaying = false
+    private let player: AVPlayer
+    private let videoOutput: AVPlayerItemVideoOutput
+    private var displayLink: CADisplayLink!
+
+    private let observers = ObservableVideoSource()
+    private var notificationToken: NSObjectProtocol?
+
+    private let queue = DispatchQueue(label: "com.mapbox.VideoPlayer")
```

Comment:
It's a good practice to include type of queue in the name.

Reply:
Code style issue. Will discuss in dedicated session.